src/java/org/apache/cassandra/index/Index.java

```diff
@@ -426,6 +426,16 @@ public Indexer indexerFor(DecoratedKey key,
                               WriteContext ctx,
                               IndexTransaction.Type transactionType);
 
+    /**
+     * Tells whether this index supports replica fitering protection or not.
+     *
+     * @return true if this index supports replica filtering protection, false otherwise
+     */
+    default boolean supportsReplicaFilteringProtection()
```

Comment:
I'd move down the new method from the [`Update processing`](https://github.com/apache/cassandra/blob/bc9a554c29248ed7733608b995cc26f410e2ba78/src/java/org/apache/cassandra/index/Index.java#L405) section to the [`Querying`](https://github.com/apache/cassandra/blob/bc9a554c29248ed7733608b995cc26f410e2ba78/src/java/org/apache/cassandra/index/Index.java#L537) section, probably right below `validate(ReadCommand)`.